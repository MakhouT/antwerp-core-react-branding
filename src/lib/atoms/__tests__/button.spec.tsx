import * as React from "react";
import * as TestUtils from "react-dom/test-utils";

import * as A from "../..";
import { renderAndMatchSnapshot, renderAsComponentAndGetByClass } from "../../../__tests__/__fixtures__/testUtils";

describe("Button", () => {
    it("Renders default button, test in prop", () => {
        // Act
        const element = (
            <A.Button
                text="testbutton"
            />);

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("Renders default button, text as child", () => {
        // Act
        const element = (
            <A.Button>testbutton</A.Button>);

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("Renders button with type and level", () => {
        // Arrang
        const level = A.Levels.Danger;
        const type = A.ButtonType.Outline;

        // Act
        const element = (
            <A.Button
                level={level}
                text="testbutton"
                type={type}
            />);

        // Act and assert
        const tree = renderAndMatchSnapshot(element);

        // Assert
        expect(tree.props.className).toContain(`a-button--${level}`);
        expect(tree.props.className).toContain(`a-button-${type}`);
        expect(tree).toMatchSnapshot();
    });

    it("Renders button with type and level, default should not be added", () => {
        // Arrang
        const level = A.Levels.Danger;
        const type = A.ButtonType.Default;

        // Act
        const element = (
            <A.Button
                level={level}
                text="testbutton"
                type={type}
            />);

        // Act and assert
        const tree = renderAndMatchSnapshot(element);

        // Assert
        expect(tree.props.className).toContain(`a-button--${level}`);
        expect(tree.props.className).not.toContain(`a-button-${type}`);
        expect(tree).toMatchSnapshot();
    });

    it("Button passes click", () => {
        // Arrang
        const onClick = jest.fn();

        // Act
        const button = renderAsComponentAndGetByClass((
            <A.Button onClick={onClick} text="mlkj" />
        ), "a-button");
        TestUtils.Simulate.click(button);

        // Assert
        expect(onClick).toBeCalled();
    });
});
