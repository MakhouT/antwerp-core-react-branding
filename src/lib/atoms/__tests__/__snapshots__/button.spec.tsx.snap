// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders button with type and level 1`] = `
<button
  autoFocus={undefined}
  className="a-button a-button-outline a-button--danger"
  disabled={undefined}
  onClick={[Function]}
  style={undefined}
  tabIndex={undefined}
  type="button"
>
  testbutton
</button>
`;

exports[`Button Renders button with type and level 2`] = `
<button
  autoFocus={undefined}
  className="a-button a-button-outline a-button--danger"
  disabled={undefined}
  onClick={[Function]}
  style={undefined}
  tabIndex={undefined}
  type="button"
>
  testbutton
</button>
`;

exports[`Button Renders button with type and level, default should not be added 1`] = `
<button
  autoFocus={undefined}
  className="a-button a-button--danger"
  disabled={undefined}
  onClick={[Function]}
  style={undefined}
  tabIndex={undefined}
  type="button"
>
  testbutton
</button>
`;

exports[`Button Renders button with type and level, default should not be added 2`] = `
<button
  autoFocus={undefined}
  className="a-button a-button--danger"
  disabled={undefined}
  onClick={[Function]}
  style={undefined}
  tabIndex={undefined}
  type="button"
>
  testbutton
</button>
`;

exports[`Button Renders default button, test in prop 1`] = `
<button
  autoFocus={undefined}
  className="a-button"
  disabled={undefined}
  onClick={[Function]}
  style={undefined}
  tabIndex={undefined}
  type="button"
>
  testbutton
</button>
`;

exports[`Button Renders default button, text as child 1`] = `
<button
  autoFocus={undefined}
  className="a-button"
  disabled={undefined}
  onClick={[Function]}
  style={undefined}
  tabIndex={undefined}
  type="button"
>
  testbutton
</button>
`;
